storage/sqlite/src/lib.rs

```diff
@@ -0,0 +1,336 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+extern crate core;
+
+mod error;
+
+use once_cell::unsync::Lazy;
+use rusqlite::{Connection, OpenFlags, OptionalExtension};
+use std::borrow::Cow;
+use std::cmp::max;
+use std::path::PathBuf;
+use std::sync::{Mutex, MutexGuard};
+use std::vec::IntoIter;
+
+use crate::error::process_sqlite_error;
+use storage_core::{
+    backend::{self, TransactionalRo, TransactionalRw},
+    info::DbDesc,
+    Data, DbIndex,
+};
+use utils::shallow_clone::ShallowClone;
+use utils::sync::Arc;
+
+// Since this is a key-value store, we have a specific structure for the underlying databases,
+// where it conforms to map<index, map<key, value>>. The outer map size almost never needs to
+// change, and hence we fix it and we use it as a constant
+const MAX_DB_COUNT: usize = 64;
+const DB_NAMES: Lazy<Vec<String>> = Lazy::new(|| make_db_names(MAX_DB_COUNT));
+
+fn make_db_names(max_size: usize) -> Vec<String> {
+    (0..max_size).map(|i| format!("db_{}", i)).collect()
+}
```

Comment:
I am not convinced pre-calculating DB names into a global variable with an arbitrary upper bound like this is a good approach.

An alternative is to pre-calculate the names when the database is open. The information about the number of key-value maps is already available at that point. The result can be stored in `DbList` and accessed by indexing a `DbIndex` into it. It can stored in a struct together with connection so the transactions can access it.

The structure of `SqliteImpl` would then be something like (excuse the made up syntax):
```rust
Arc<{ connection: Mutex<sqlite::Connection>, db_names: Vec<String> }>
```

Reply:
That sounds better. Let's go for it! 

@erasmospunk could you give this a shot? I think all you have to do is eliminate the global variable, the once_cell dependency, and put the implementation inside open_db(), and store the values in the struct. Please give it a shot.

Reply:
Implemented your idea as `SqliteQueries`